### PR TITLE
Portals 560

### PR DESCRIPTION
**Pull request by conundrumgirl**
**DO NOT MERGE TILL MONDAY** 
- Changes for UserFileGrid and the upload tool to accommodate the new FormData API
- Save file and Submit functionality
- Enhancing the 'warning when navigating away'

### CODE REVIEW

File: src/__tests__/lib/containers/drug_upload_tool/DrugUploadTool.test.tsx [hunk 1]
```diff
@@ -44,99 +47,196 @@ const props: DrugUploadToolProps = {
...
+        .props() as any) as DrugUploadFormProps
+      expect(formProps.formTitle).toBe('Another Title')
+      expect(formProps.formData).toEqual({})
+      expect(formProps.isWizardMode).toBeFalsy()
+    })
+  })
+})
+
+describe('saving data file', () => {
+  it('should create formDtaa if there is not and ID', async () => {
```
Comment (leem42) [inline src/__tests__/lib/containers/drug_upload_tool/DrugUploadTool.test.tsx]: formDtaa

File: src/__tests__/lib/containers/drug_upload_tool/DrugUploadTool.test.tsx [hunk 2]
```diff
@@ -44,99 +47,196 @@ const props: DrugUploadToolProps = {
...
+      Promise.resolve(JSON.stringify(formschemaJson)),
+    ))
+    const formData = _.cloneDeep(formschemaJson)
+    _.set(formData, props.fileNamePath, 'someName')
+    await instance.saveToFile(formData)
+    expect(create).toHaveBeenCalled()
+    expect(update).not.toHaveBeenCalled()
+  })
+
+  it('should create formDtaa if there is not and ID', async () => {
```
Comment (leem42) [inline src/__tests__/lib/containers/drug_upload_tool/DrugUploadTool.test.tsx]: relabeled

File: src/__tests__/lib/containers/drug_upload_tool/DrugUploadTool.test.tsx [hunk 3]
```diff
@@ -44,99 +47,196 @@ const props: DrugUploadToolProps = {
...
+    const save = jest.spyOn(instance, 'saveToFile')
+
+    const formData = _.cloneDeep(formschemaJson)
+    _.set(formData, props.fileNamePath, 'someName')
+    await instance.submitForm(formData)
+    expect(save).toHaveBeenCalled()
+    expect(submit).toHaveBeenCalled()
+  })
+
+  it('should create formDtaa if there is not and ID', async () => {
```
Comment (leem42) [inline src/__tests__/lib/containers/drug_upload_tool/DrugUploadTool.test.tsx]: delete

File: src/lib/containers/drug_upload_tool/DrugUploadForm.tsx [hunk 4]
```diff
@@ -111,19 +119,37 @@ export default class DrugUploadForm extends React.Component<
...
+  setupBeforeUnloadListener = () => {
+    window.addEventListener('beforeunload', this.onUnload)
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('beforeunload', this.onUnload)
+  }
+
+  componentDidUpdate(prevProps: DrugUploadFormProps) {
+    const shouldUpdate = this.props.callbackStatus !== prevProps.callbackStatus
```
Comment (leem42) [inline src/lib/containers/drug_upload_tool/DrugUploadForm.tsx]: check if callback status is succesfull

File: src/lib/containers/drug_upload_tool/DrugUploadTool.tsx [hunk 5]
```diff
@@ -1,270 +1,242 @@
...
-      status: StatusEnum.SUCCESS
-    });
+      notification: { type: status, message: message },
+      status: status,
+    })
     setTimeout(() => {
-      this.setState({ status: undefined });
-    }, 5000);
-  };
+      this.setState({ status: undefined })
```
Comment (leem42) [inline src/lib/containers/drug_upload_tool/DrugUploadTool.tsx]: short time

File: src/lib/containers/drug_upload_tool/DrugUploadTool.tsx [hunk 6]
```diff
@@ -1,270 +1,242 @@
...
       },
       status: StatusEnum.ERROR,
-      isLoading: false
-    });
-  };
+      isLoading: false,
+    })
+  }
+
+  /* alina TODO! This will change*/
```
Comment (leem42) [inline src/lib/containers/drug_upload_tool/DrugUploadTool.tsx]: can remove

File: src/lib/containers/drug_upload_tool/UserFileGrid.tsx [hunk 7]
```diff
@@ -84,110 +81,52 @@ export default class UserFileGrid extends React.Component<
...
-        parentId: parentContainerId,
+      const request: ListRequest = {
+        filterByState: [StatusEnum.WAITING_FOR_SUBMISSION],
+        groupId
+        // nextPageToken?: string
       }
-      const entity = await SynapseClient.lookupChildEntity(
-        entityLookupRequest,
+
+      const list: ListResponse = await SynapseClient.listFormData(
```
Comment (leem42) [inline src/lib/containers/drug_upload_tool/UserFileGrid.tsx]: https://github.com/Sage-Bionetworks/Synapse-React-Client/blob/91b8def2a4a947567a4f85cfbafa951b24b6cc3e/src/lib/utils/SynapseClient.ts#L349

**Comment by conundrumgirl** [comment]
@jay-hodgson  - just FYI

**Pull request merged**